server/CGameHandler.cpp

```diff
@@ -1029,7 +1029,7 @@ void CGameHandler::makeAttack(const CStack * attacker, const CStack * defender,
 		}
 		if (VLC->modh->settings.data["hardcodedFeatures"]["NEGATIVE_LUCK"].Bool()) // negative luck enabled
 		{
-			if (attackerLuck < 0 && getRandomGenerator().nextInt(23) < abs(attackerLuck))
+			if (attackerLuck != INT_MIN && attackerLuck < 0 && getRandomGenerator().nextInt(23) < abs(attackerLuck))
```

Comment:
Why luck can be INT_MIN? :)

Reply:
probably abs(INT_MIN) gives something bad and this is why it triggered a warning.

Reply:
`abs(INT_MIN)` would be an overflow as it would try to fit the value in `int`

Reply:
Noise, would never happen:
`const int attackerLuck = attacker->LuckVal();`
And LuckVal can only return values between -3 and +3